garak/generators/llm.py

```diff
@@ -0,0 +1,120 @@
+# SPDX-FileCopyrightText: Portions Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+"""LLM (simonw/llm) generator support"""
+
+import logging
+from typing import List, Union
+
+import llm 
+
+from garak import _config
+from garak.attempt import Message, Conversation
+from garak.generators.base import Generator
+
+
+class LLMGenerator(Generator):
+    """Class supporting simonw/llm-managed models
+
+    See https://pypi.org/project/llm/ and its provider plugins.
+
+    Calls model.prompt() with the prompt text and relays the response. Per-provider
+    options and API keys are all handled by `llm` (e.g., `llm keys set openai`).
+
+    Set --model_name to the `llm` model id or alias (e.g., "gpt-4o-mini",
+    "claude-3.5-haiku", or a local alias configured in `llm models`).
+
+    Explicitly, garak delegates the majority of responsibility here:
+
+    * the generator calls prompt() on the resolved `llm` model
+    * provider setup, auth, and model-specific options live in `llm`
+    * there's no support for chains; this is a direct LLM interface
+
+    Notes:
+    * Not all providers support all parameters (e.g., temperature, max_tokens).
+      We pass only non-None params; providers ignore what they don't support.
+    """
+
+    DEFAULT_PARAMS = Generator.DEFAULT_PARAMS | {
+        "max_tokens": None,
+        "top_p": None,
+        "stop": [],
+    }
+
+    generator_family_name = "llm"
+
+    def __init__(self, name: str = "", config_root=_config):
+        self.target = None
+        self.name = name
+        self._load_config(config_root)
+        self.fullname = f"llm (simonw/llm) {self.name or '(default)'}"
+
+        self._load_client()
+
+        super().__init__(self.name, config_root=config_root)
+
+        self._clear_client()
```

Comment:
It may be best to call `super().__init__()` before calling `_load_client()` to ensure the end object state is not impacted. Also no need to call `_clear_client()` during `__init__()`.
``` Suggestion
        super().__init__(self.name, config_root=config_root)
        self._load_client()
```